.github/workflows/build_and_test.yml

```diff
@@ -118,20 +138,38 @@ jobs:
       - name: Install dependencies
         run: |
           sudo apt-get update -y -qq
-          sudo apt-get install -y -qq build-essential clang curl ninja-build libgmp-dev libmpfr-dev gcc${{ matrix.package || format('-{0}-linux-gnu', matrix.arch) }}
+          sudo apt-get install -y -qq build-essential curl ninja-build libgmp-dev libmpfr-dev gcc${{ matrix.gnupkg || format('-{0}-linux-gnu', matrix.arch) }}
+
+      - name: Install gcc
+        run: |
+          RISCV_GNU_TOOLCHAIN_TAG="2023.11.08"
+          curl -L https://github.com/riscv-collab/riscv-gnu-toolchain/releases/download/${RISCV_GNU_TOOLCHAIN_TAG}/riscv64-glibc-ubuntu-20.04-gcc-nightly-${RISCV_GNU_TOOLCHAIN_TAG}-nightly.tar.gz |
+            tar xzf - -C /opt
+          echo "PATH=/opt/riscv/bin:$PATH" >> $GITHUB_ENV
+        if: ${{ matrix.compiler == 'gcc' && matrix.arch == 'riscv64' }}
```

Comment:
When comes the time to bump to gcc 14 in order to activate test for riscv64, there is a chance we use a more straightforward build of gcc (as in more widely used). For the sake of clarity would you mind removing this step, or replacing the content by 
```
sudo apt-get install -qq gcc${{ matrix.gnupkg || format('-{0}-linux-gnu', matrix.arch) }}
```
for all architectures so it matches the name of the step better and we don't apt install gcc at all for riscv64 (for now).